src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/formLayout.ts

```diff
@@ -21,3 +22,27 @@ export function getLayoutElementIndexById(elementId: string, formLayout: [ILayou
   }
   return formLayout.findIndex((element) => element.id === elementId);
 }
+
+export function getRepeatingGroups(formLayout: [ILayoutComponent | ILayoutGroup], formData: any) {
+  const repeatingGroups: IRepeatingGroups = {};
+  formLayout.filter((layoutElement) => layoutElement.type === 'group')
+    .forEach((groupElement: ILayoutGroup) => {
+      if (groupElement.maxCount > 1) {
+        const groupFormData = Object.keys(formData).filter((key) => {
+          return key.startsWith(groupElement.dataModelBindings.group);
+        });
+        if (groupFormData && groupFormData.length > 0) {
+          const lastItem = groupFormData[groupFormData.length - 1];
+          const regex = new RegExp(/\[([0-9]+)\]/);
```

Comment:
Should regexp be moved outside the forEach to avoid recreating the same object for each pass?

Reply:
Agreed, fixed.